### PR TITLE
[feature](merge-cloud) cloud meta mgr impl prepare/commit/sync rowset

### PR DESCRIPTION
**Pull request by w41ter**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by w41ter** [comment]
run buildall

File: be/src/cloud/cloud_meta_mgr.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ class CloudMetaMgr {
 private:
     Status sync_tablet_delete_bitmap(
             CloudTablet* tablet, int64_t old_max_version,
-            const google::protobuf::RepeatedPtrField<RowsetMetaPB>& rs_metas,
```
Comment (platoneko) [inline be/src/cloud/cloud_meta_mgr.h:91]: Consider range view

File: be/src/olap/base_tablet.h [hunk 2]
```diff
@@ -88,7 +89,34 @@ class BaseTablet {
...
+
+    // operation for clone
+    void calc_missed_versions(int64_t spec_version, std::vector<Version>* missed_versions);
+    void calc_missed_versions_unlocked(int64_t spec_version,
+                                       std::vector<Version>* missed_versions) const;
+
+    size_t num_rows();
+    int version_count() const;
+    uint64_t segment_count() const;
+    Version max_version() const;
```
Comment (platoneko) [inline be/src/olap/base_tablet.h:115]: `CloudTablet` 有一些已经聚合的状态值可以直接用在这

File: be/src/olap/base_tablet.cpp [hunk 3]
```diff
@@ -119,4 +118,149 @@ bool BaseTablet::_reconstruct_version_tracker_if_necessary() {
...
+
+void BaseTablet::calc_missed_versions(int64_t spec_version, std::vector<Version>* missed_versions) {
+    std::shared_lock rdlock(_meta_lock);
+    calc_missed_versions_unlocked(spec_version, missed_versions);
+}
+
+// for example:
+//     [0-4][5-5][8-8][9-9]
+// if spec_version = 6, we still return {7} other than {6, 7}
+void BaseTablet::calc_missed_versions_unlocked(int64_t spec_version,
```
Comment (platoneko) [inline be/src/olap/base_tablet.cpp:234]: `CloudTablet` 好像已经有同名不同实现类似语义的函数

File: be/src/olap/base_tablet.cpp [hunk 4]
```diff
@@ -119,4 +118,149 @@ bool BaseTablet::_reconstruct_version_tracker_if_necessary() {
...
+                                               std::vector<Version>* missed_versions) const {
+    DCHECK(spec_version > 0) << "invalid spec_version: " << spec_version;
+    std::list<Version> existing_versions;
```
Comment (platoneko) [inline be/src/olap/base_tablet.cpp:237]: list sort 是不是很6

File: be/src/olap/rowset/rowset_meta.cpp [hunk 5]
```diff
@@ -102,18 +102,20 @@ void RowsetMeta::set_fs(io::FileSystemSPtr fs) {
...
+        if (!skip_schema) {
+            // For cloud, separate tablet schema from rowset meta to reduce persistent size.
+            _schema->to_schema_pb(rs_meta_pb->mutable_tablet_schema());
+        }
     }
 }
 
-RowsetMetaPB RowsetMeta::get_rowset_pb() {
+RowsetMetaPB RowsetMeta::get_rowset_pb(bool skip_schema) const {
     RowsetMetaPB rowset_meta_pb = _rowset_meta_pb;
```
Comment (gavinchou) [inline be/src/olap/rowset/rowset_meta.cpp:117]: seems unnecessary to assign init.
```suggestion
    RowsetMetaPB rowset_meta_pb;
```

**Comment by w41ter** [comment]
run buildall

**Comment by w41ter** [comment]
run buildall

**Comment by w41ter** [comment]
run buildall

**Comment by w41ter** [comment]
run buildall

**Review by platoneko — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**